### PR TITLE
Redirect to asset page after item is placed on sale

### LINKED ISSUE #496: Redirect to asset page after item is placed on sale
Screen Shot 2019-05-29 at 3 35 20 PM

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #496

### CODE REVIEW

File: src/components/common/steps_modal/buy_sell_collectible_step.tsx [hunk 1]
```diff
@@ -101,6 +113,11 @@ class BuySellCollectibleStep extends React.Component<Props, State> {
                 onLoading();
                 await this.props.submitCollectibleOrder(signedOrder);
                 onDone();
+
```
Comment (Agupane) [inline src/components/common/steps_modal/buy_sell_collectible_step.tsx:113]: R:
With the current logic, the user will be redirected to the collectibles page without the modal to be closed. I would add a "modalClose" function here before redirecting.

I think you can pass ` closeModal={reset}` to this modal as props  from `/src/components/common/steps_modals/steps_modals.tsx` and use that to close the modal before redirecting.

File: src/components/common/steps_modal/buy_sell_collectible_step.tsx [hunk 2]
```diff
@@ -101,6 +113,11 @@ class BuySellCollectibleStep extends React.Component<Props, State> {
...
+                // Go to the collectible's profile
+                if (!currentRoutePath.includes(`collectible/${collectible.tokenId}`)) {
```
Comment (fvictorio) [inline src/components/common/steps_modal/buy_sell_collectible_step.tsx]: I think we need to stop doing this. We are hardcoding partial URLs everywhere and relaying a lot on this pattern of `currentRoutePath.includes`. It smells bad to me.

I don't know how is this supposed to be done the right way in `react-router`, but there must be a better way.

Comment (unjapones) [reply src/components/common/steps_modal/buy_sell_collectible_step.tsx]: Maybe we can dispatch [`goToIndividualCollectible`](https://github.com/0xProject/0x-launch-kit-frontend/blob/development/src/store/router/actions.ts#L70) and move this verification there?

To me, it seems that the condition check on the redirect is not a responsability of this component/file.

**Review by Agupane — CHANGES_REQUESTED** [review]

File: src/components/common/steps_modal/buy_sell_collectible_step.tsx (hunk 1 above)
Comment (mariano-aguero) [reply src/components/common/steps_modal/buy_sell_collectible_step.tsx:113]: Done in this [commit](https://github.com/0xProject/0x-launch-kit-frontend/pull/502/commits/64d5c2be5b301f3850e794aba827f4746b21edb9)

File: src/components/common/steps_modal/buy_sell_collectible_step.tsx (hunk 2 above)
Comment (mariano-aguero) [reply src/components/common/steps_modal/buy_sell_collectible_step.tsx]: > Maybe we can dispatch [`goToIndividualCollectible`](https://github.com/0xProject/0x-launch-kit-frontend/blob/development/src/store/router/actions.ts#L70) and move this verification there?
> 
> To me, it seems that the condition check on the redirect is not a responsability of this component/file.

@unjapones  Done in this [commit](https://github.com/0xProject/0x-launch-kit-frontend/pull/502/commits/bc6b4686fcbe044c926f617111d27c157517a598)

**Review by fvictorio — APPROVED** [review]

**Review by Agupane — APPROVED** [review]

**Pull request merged**